### DISCUSSION TITLE
dss2 windows 11 error

### DISCUSSION THREAD

**Discussion by ytoaa** [staxrip/staxrip#923] [Q&A]
I'm using that encoder well
If you try to encode the file by setting it to dss2 in Windows 11, the error will occur.
Surprisingly, it works well when you do the same project on Windows 10.
I don't know why. Please help me

Script Error

DSS2: Can't open "C:\Users\nawor\Downloads\test.mp4"

Connect filters (Src_Out + VS_In): (80040217) 연결을 가능하게 하는 중간 필터들을 찾을 수 없습니다.

(C:\Users\nawor\Downloads\test_temp\test.avs, line 3)

**Comment by jessielw** [comment]
I don't know if this is helpful. However, it looks like that mod hasn't been updated since **2014/10/05**. There is a good chance it wasn't designed to work with Windows 11. 

Have you tried it separate from StaxRip in an editor?

**Reply by ytoaa** [reply]
Where is the editor?

**Reply by jessielw** [reply]
You should use something like AvsPmod. 

https://github.com/gispos/AvsPmod/releases

**Reply by ytoaa** [reply]
Unfortunately, I don't know how to use it.

**Reply by jessielw** [reply]
It has a little bit of a learning curve. 

In StaxRip if you select preview code (in the avs section on the left of the main GUI), then copy all that code, paste it into the AvsPmod ediotor, select F5, it should work. 

EDIT: 
I'd test this for you but I won't be upgrading to Windows 11 until I upgrade my computer at the start of the year

**Reply by ytoaa** [reply]
![화면 캡처 2022-08-04 002522](https://user-images.githubusercontent.com/24527825/182647008-b1145896-8c7b-43a4-9ee3-86c58eeb8283.png)
Unfortunately, the same error came up It's very sad.

**Reply by jessielw** [reply]
That let's you know the filter just won't work with Windows 11. 

Now, I don't know your specific use case, but is there a reason why you're not using FFMS or L-Smash?

**Reply by ytoaa** [reply]
When you test the video with something else, a certain part of the video suddenly skips or passes quickly at a strange speed. Only dss2 was able to encode properly.

**Reply by jessielw** [reply]
That was most likely with older versions of FFMS.

What is the source file in question? The container/codec used? 

The entire point of using source filters is to manipulate it frame by frame. However, when x264/x265 encodes it goes frame by frame regardless. So the skipping would be a non issue. 

L-Smash is 100% frame accurate/up to date. FFMS can still _sometimes_ have skips, but again it's a non issue.

**Reply by ytoaa** [reply]
Wait a minute. I'll upload the problem part of the video.

**Reply by jessielw** [reply]
Just upload the MediaInfo of said source file

**Reply by ytoaa** [reply]
General
Complete name                  : C:\Users\nawor\Downloads\test.mp4
Format                         : MPEG-4
Format profile                 : Base Media / Version 2
Codec ID                       : mp42 (mp42/isom/avc1)
File size                      : 145 MiB
Duration                       : 27 min 39 s
Overall bit rate mode          : Variable
Overall bit rate               : 733 kb/s
Encoded date                   : UTC 2012-02-17 03:10:20
Tagged date                    : UTC 2012-02-17 03:12:31
Writing application            : HandBrake 0.9.5 2011010300

Video
ID                             : 1
Format                         : AVC
Format/Info                    : Advanced Video Codec
Format profile                 : Main@L3
Format settings                : CABAC / 4 Ref Frames
Format settings, CABAC         : Yes
Format settings, Reference fra : 4 frames
Codec ID                       : avc1
Codec ID/Info                  : Advanced Video Coding
Duration                       : 27 min 38 s
Bit rate                       : 573 kb/s
Width                          : 854 pixels
Height                         : 480 pixels
Display aspect ratio           : 16:9
Frame rate mode                : Variable
Frame rate                     : 14.511 FPS
Minimum frame rate             : 2.500 FPS
Maximum frame rate             : 23.981 FPS
Color space                    : YUV
Chroma subsampling             : 4:2:0
Bit depth                      : 8 bits
Scan type                      : Progressive
Bits/(Pixel*Frame)             : 0.096
Stream size                    : 113 MiB (78%)
Writing library                : x264 core 112
Encoding settings              : cabac=1 / ref=2 / deblock=1:0:0 / analyse=0x1:0x111 / me=hex / subme=6 / psy=1 / psy_rd=1.00:0.00 / mixed_ref=0 / me_range=16 / chroma_me=1 / trellis=0 / 8x8dct=0 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-2 / threads=12 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / constrained_intra=0 / bframes=2 / b_pyramid=2 / b_adapt=1 / b_bias=0 / direct=1 / weightb=0 / open_gop=0 / weightp=2 / keyint=240 / keyint_min=23 / scenecut=40 / intra_refresh=0 / rc_lookahead=40 / rc=crf / mbtree=1 / crf=21.0 / qcomp=0.60 / qpmin=3 / qpmax=51 / qpstep=4 / ip_ratio=1.40 / aq=1:1.00
Encoded date                   : UTC 2012-02-17 03:10:20
Tagged date                    : UTC 2012-02-16 22:03:10
Color range                    : Limited
Color primaries                : BT.601 NTSC
Transfer characteristics       : BT.709
Matrix coefficients            : BT.601
Codec configuration box        : avcC

Audio
ID                             : 2
Format                         : AAC LC
Format/Info                    : Advanced Audio Codec Low Complexity
Codec ID                       : mp4a-40-2
Duration                       : 27 min 39 s
Bit rate mode                  : Variable
Bit rate                       : 154 kb/s
Maximum bit rate               : 347 kb/s
Channel(s)                     : 2 channels
Channel layout                 : L R
Sampling rate                  : 44.1 kHz
Frame rate                     : 43.066 FPS (1024 SPF)
Compression mode               : Lossy
Stream size                    : 30.6 MiB (21%)
Title                          : Stereo
Encoded date                   : UTC 2012-02-17 03:10:20
Tagged date                    : UTC 2012-02-16 22:03:10

**Reply by jessielw** [reply]
Ouch that's really **low bitrate** to encode a video from!

Never the less, LWLibavVideoSource (L-Smash) works perfectly with AVC formats. I recommend you swap to it and use FFMS as a back up on the rare codec L-Smash does not work well on.

**Reply by ytoaa** [reply]
I tested it again, but if it's not dss2, the problem will pass at an abnormal speed.

**Reply by jessielw** [reply]
You have some major issues with this source file tbh. The FPS is 14.511. 

You could also use LSmashVideoSource for .mp4 files.

**Reply by ytoaa** [reply]
I tested it, but it's the same symptom.

**Reply by jessielw** [reply]
My next advice, is to extract the actual video files from the container and try a well known source on it (ffms/lsmash)

**Reply by ytoaa** [reply]
Oh, I knew this was a different matter. There was a problem with Intel qsv encoding. When it comes to "copy/mux", it goes without skipping, but when it comes to "intel x265", the Halsey video is fast
Skipped or fast.

**Reply by jessielw** [reply]
So if you used software encoding there is no issue?

**Reply by ytoaa** [reply]
Well, it's the same symptom with software encoding.

**Reply by jessielw** [reply]
Did you try to extract the actual raw video file? 

This just seems like the frame rate is being incorrectly detected. This could solve your issue.

**Reply by ytoaa** [reply]
Only encoding through dss2 proceeds without image skip
How do you extract raw video?

**Reply by jessielw** [reply]
For MP4 you'd need to use this command with FFMPEG. 

`ffmpeg -i test.mp4 -vcodec copy -an -bsf:v h264_mp4toannexb test.h264`

**Reply by ytoaa** [reply]
Is there any way to change the frame speed detection with the staxrip setting?

**Reply by jessielw** [reply]
Please post your avs script (the script you copied into avspmod)

**Reply by jessielw** [reply]
Under your source filter add `ConvertFPS(new_rate=23.976)`

```
ffvideosource(blah blah)
ConvertFPS(new_rate=23.976)

rest of script
```

**Reply by ytoaa** [reply]
When using convertfps, there is a problem that the image is too long in a certain interval.
I think just using dss2 on Windows 10 will be good for mental health.

**Reply by jessielw** [reply]
Maybe someone with a little more knowledge can help you. If I was you I'd post on videohelp forum for the correct solution. Using an out of date source filter might work, but it's definitely not a long term solution. Regardless, this isn't an issue with StaxRip.

**Reply by ytoaa** [reply]
Thank you for your answer.

**Reply by ytoaa** [reply]
Oh... I unexpectedly figured out the cause. During encoding setup, the skip phenomenon disappeared when the decoder was changed from "avisynth/vapoursynth" to something else.

**Reply by jessielw** [reply]
What did you select? I guess the pipe was causing an issue.

**Reply by ytoaa** [reply]
I tested "ffmpeg dxva2, qsv hardware" These two became normal.

**Reply by ytoaa** [reply]
The preset is intel x265.

**Reply by jessielw** [reply]
I'm not really sure why this would cause an issue. I'm assuming the pipe that is feeding avisynth/vs correctly. This is beyond the scope of my knowledge though.

**Reply by ytoaa** [reply]
The frame rate during "input/output" setting in software encoding seems to be causing problems I don't think the variable scan rate of the image is properly recognized

**Reply by ytoaa** [reply]
There was a problem with the pipe "none" in the software encoding.

**Reply by jessielw** [reply]
The frame rate in the mediainfo is dropping from 2 frames to 24. So, I'm really unsure if this is a pipe error or source error tbh.

**Reply by ytoaa** [reply]
Haha... x265 software encoding is successful when you replace the decoder with "ffmpeg dxva2"

**Reply by jessielw** [reply]
So pretty much when directly using FFMPEG as the decoder, to feed the encoder, it works good. This again, is probably an issue with the pipes that staxrip uses to send the file to the encoders.

We would need @Dendraspis and possibly @Patman86 to solve this. @Dendraspis is the maintainer, but he has been missing a little while.

**Reply by Patman86** [reply]
DSS2 need Haali Media Splitter or LAV Filters to Work. Have you installed one of them?

**Reply by ytoaa** [reply]
Do I just install it separately?

**Reply by jessielw** [reply]
Nice!

**Reply by Patman86** [reply]
> Do I just install it separately?

Yes.
